doc/release-notes.md

```diff
@@ -23,492 +26,644 @@ shut down (which might take a few minutes for older versions), then run the
+and transition toward the new end-state allocation with minimal network disruption.
+The proposed changes will occur every three superblock cycles until the reallocation
+is complete, or approximately once per quarter.
+
+Note that this is a hardfork which must be activated by miners. To do this they
+should start creating blocks signalling bit 5 in `version` field of the block header.
+
+Dynamic Activation Thresholds
+-----------------------------
+In Dash we used to use lower thresholds (80% vs 95% in BTC) to activate upgrades
+via BIP9-like mechanism for quite some time. While it's preferable to have as much
+of the network hashrate to signal update readiness as possible this can result in
+quite lengthy upgrades sometimes when one large non-upgraded entity would stale
+the whole progress. Simply lowering thresholds even further can result in network
+upgrades being too fast which can cause some chaos potentially. This version
+implements BIP9-like dynamic activation thresholds which drop from some initial
+level to a minimally acceptable one over time at an increasing rate providing
+a safe non-blocking way of activating proposals.
+
+This mechanism applies to the Block Reward Reallocation proposal mentioned above.
+
+Concentrated Recovery
+---------------------
+In the current system, signature shares are propagated to all LLMQ members,
+until one of them has enough shares collected to recover the signature. Until
+this recovered signature is propagated in the LLMQ, all members will keep
+propagating shares and verifying each one. This causes quite some load on the
+LLMQ, which will be avoided with the new system.
+
+The new system sends all shares to only a single deterministically selected node,
+so that this node can recover the signature and propagate the recovered signature.
+This way only the recovered signature needs to be propagated and verified by all
+members. After sending the share to this single node, each member waits for some
+time and repeats sending it to another deterministically selected member. This
+process is repeated until a recovered signature is finally created and propagated.
+
+The new system is activated with the newly added `SPORK_21_QUORUM_ALL_CONNECTED`.
+
+Increased number of masternode connections
+------------------------------------------
+To implement "Concentrated Recovery", it is now required that all members of a LLMQ
+connect to all other members of the same LLMQ. This increases general connection count
+for masternodes a lot. These intra-quorum connections are less resource consuming than
+normal p2p connections as they only exchange LLMQ/masternode related messages, but
+the hardware and network requirements will still be higher than before.
+
+This change will at first only be activated for the smaller LLMQs (50 members) and
+then later for the larger ones (400 members).
+
+This is also controlled via `SPORK_21_QUORUM_ALL_CONNECTED`.
```

Comment:
Should talk about the duality of the spork here? difference between 0 being all quorums 1 being small quorums?